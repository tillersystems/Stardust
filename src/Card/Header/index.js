import PropTypes from 'prop-types';
import styled, { css } from 'styled-components';

/**
 * Header
 *
 * This component is in charge of display
 * the header of Card component
 *
 * @param {boolean} small - change header size
 *
 * @return {jsx}
 */
const Header = styled.div`
  border-bottom: solid 1px ${({ theme: { palette } }) => palette.veryLightBlue};

  display: flex;
  align-items: center;
  justify-content: space-between;

  height: 7.4rem;
  padding: 0 3rem;

  ${({ small }) =>
    small &&
    css`
      height: 6rem;
      padding: 0 2rem;
    `}
`;

/**
 * PropTypes Validation
 */
const { bool } = PropTypes;
Header.propTypes = {
  small: bool,
};

/**
 * Default props
 */
Header.defaultProps = {
  small: false,
};

export default Header;
