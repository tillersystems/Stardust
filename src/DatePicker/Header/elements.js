import styled from 'styled-components';

import { Button } from '../../..';

export const Title = styled.div`
  flex: 1;

  user-select: none;

  font-weight: ${({ theme: { fonts } }) => fonts.weight.thick};
  font-size: ${({ theme: { fonts } }) => fonts.size.medium};
`;

export const HeaderButton = styled(Button)`
  padding: 0;
`;
