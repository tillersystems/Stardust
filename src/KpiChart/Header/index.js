import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

const KpiChartHeader = ({ children, className }) => <div className={className}>{children}</div>;

/**
 * PropTypes Validation
 */
const { node, string } = PropTypes;
KpiChartHeader.propTypes = {
  children: node,
  className: string,
};

/**
 * Default props
 */
KpiChartHeader.defaultProps = {
  children: null,
  className: '',
};

export default styled(KpiChartHeader)`
  display: flex;
  justify-content: space-between;
  align-items: center;
`;
