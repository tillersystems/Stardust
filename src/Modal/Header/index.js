import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

const ModalHeader = ({ children, className }) => <div className={className}>{children}</div>;

/**
 * PropTypes Validation
 */
const { node, string } = PropTypes;
ModalHeader.propTypes = {
  children: node,
  className: string,
};

/**
 * Default props
 */
ModalHeader.defaultProps = {
  children: null,
  className: '',
};

export default styled(ModalHeader)`
  height: 6rem;
  border-bottom: 0.1rem solid ${({ theme: { palette } }) => palette.gray};
  width: 100%;
  display: flex;
  align-items: center;
  justify-content: space-between;
`;
