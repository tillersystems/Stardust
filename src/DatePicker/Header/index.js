import React, { memo } from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

import Icon from '../../Icon';
import Theme from '../../Theme';
import { Title, HeaderButton } from './elements';

/**
 * Header
 *
 * This component is in charge of displaying date picker header.
 *
 * @param {string} className - Class needed by styled component.
 * @param {string} title - The title of the header.
 * @param {func} prev - Prev button handler.
 * @param {func} next - Next button handler.
 * @param {bool} disablePrev - Indicates if the prev button should be disabled.
 * @param {bool} disableNext - Indicates if the next button should be disabled.
 * @param {number} monthIndex - Month index ( rely on numberOfMonthsToDisplay).
 *
 * @return {jsx}
 */
const Header = ({ className, title, prev, next, disablePrev, disableNext, monthIndex }) => (
  <div className={className}>
    <HeaderButton
      data-testid="previous-month-button"
      size="small"
      onClick={() => prev(monthIndex)}
      disabled={disablePrev}
    >
      <Icon color={Theme.palette.spaceGrey} name="chevron-left" />
    </HeaderButton>

    <Title>{title}</Title>

    <HeaderButton
      data-testid="next-month-button"
      size="small"
      onClick={() => next(monthIndex)}
      disabled={disableNext}
    >
      <Icon color={Theme.palette.spaceGrey} name="chevron-right" />
    </HeaderButton>
  </div>
);

/**
 * PropTypes Validation.
 */
const { bool, func, number, string } = PropTypes;
Header.propTypes = {
  className: string,
  title: string,
  prev: func,
  monthIndex: number,
  next: func,
  disablePrev: bool,
  disableNext: bool,
};

/**
 * Default props.
 */
Header.defaultProps = {
  className: '',
  title: '',
  prev: () => {},
  monthIndex: 0,
  next: () => {},
  disablePrev: false,
  disableNext: false,
};

const styledHeader = styled(Header)`
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  align-items: center;

  color: ${({ theme: { palette } }) => palette.darkBlue};
  text-align: center;
  font-size: ${({
    theme: {
      fonts: { size },
    },
  }) => size.default};

  line-height: ${({
    theme: {
      fonts: { size },
    },
  }) => size.medium};
`;

styledHeader.displayName = 'Header';

export default memo(styledHeader);
