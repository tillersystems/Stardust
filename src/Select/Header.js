import React, { Children } from 'react';
import PropTypes from 'prop-types';

import { HeaderButton, HeaderContent } from './elements';

/**
 * SearchBar
 *
 * This component is in charge of displaying
 * a search bar
 *
 * @param {bool} disabled - If button is disabled
 * @param {function} onClick - Callback function called when button is clicked
 * @param {bool} isOpen - When popover is open
 * @param {string} placeholder - Placeholder of button when no value is selected
 * @param {node} children - The content of Header
 *
 * @return {jsx}
 */
const Header = ({ disabled, onClick, isOpen, placeholder, children }) => {
  // Remove empty children to ensure placeholder is display if empty
  const filteredChildren = Children.toArray(children).filter(Boolean);

  return (
    <HeaderButton
      type="button"
      disabled={disabled}
      onClick={onClick}
      aria-haspopup="true"
      aria-expanded={isOpen}
      data-testid="select-button"
    >
      <HeaderContent>{filteredChildren.length ? filteredChildren : placeholder}</HeaderContent>
    </HeaderButton>
  );
};

const { bool, func, node, string } = PropTypes;

/** Prop types. */
Header.propTypes = {
  children: node,
  disabled: bool,
  isOpen: bool,
  onClick: func.isRequired,
  placeholder: string,
};

/** Default props. */
Header.defaultProps = {
  children: null,
  disabled: false,
  isOpen: false,
  placeholder: '',
};

export default Header;
