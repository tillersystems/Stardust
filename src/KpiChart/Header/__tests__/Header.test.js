import React from 'react';

import Header from '..';

describe('<KpiChart.Header />', () => {
  test('should render without a problem', () => {
    const { container } = render(<Header>Header</Header>);

    expect(container.firstChild).toMatchSnapshot();
  });
});
