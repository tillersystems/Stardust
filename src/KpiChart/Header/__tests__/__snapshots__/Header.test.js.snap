// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<KpiChart.Header /> should render without a problem 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

<div
  class="c0"
>
  Header
</div>
`;
